test/generator/template_values_test.rb

```diff
@@ -56,4 +56,115 @@ def teardown
       $LOAD_PATH.delete 'test/fixtures/xruby/lib'
     end
   end
+
+  class CaseValuesTest < Minitest::Test
+    def setup
+      $LOAD_PATH.unshift 'test/fixtures/xruby/lib'
+    end
+
+    def simple_canonical_data
+      simple_canonical_data = Minitest::Mock.new
+      simple_canonical_data.expect(
+        :to_s,
+        <<-TEXT
+{
+	"description": "Test canonical data",
+	"cases": [
+	{
+		"description": "add 2 numbers",
+		"input": [1,1],
+		"expected": 2
+	}
+	]
+}
+TEXT
+      )
+      simple_canonical_data
+    end
+
+    def complex_canonical_data
```

Comment:
We discussed this in gitter and have decided to try a data section at the end of the test file. 
The main thing is removing the big heredoc from the middle of the test file.